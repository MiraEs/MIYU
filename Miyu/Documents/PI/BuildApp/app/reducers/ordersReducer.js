'use strict';

import {
	LOAD_ORDERS,
	LOAD_ORDERS_SUCCESS,
	LOAD_ORDERS_FAIL,

	LOAD_ORDER_DETAILS_SUCCESS,
	LOAD_ORDER_DETAILS_FAIL,

	ADD_PROJECT_ORDER,

	LOAD_RETURNS,
	LOAD_RETURNS_SUCCESS,
	LOAD_RETURNS_FAIL,

	LOAD_RETURN_DETAILS_SUCCESS,
	LOAD_RETURN_DETAILS_FAIL,

	LOAD_SHIPPING_INFO_SUCCESS,
	LOAD_SHIPPING_INFO_FAIL,

} from '../constants/constants';
import { SIGN_USER_OUT } from '../constants/Auth';

const initialState = {
	orders: [],
	loadingOrders: true,
	orderDetails: {},
	returns: [],
	loadingReturns: true,
	returnDetails: {},
	shippingInfo: {},
	errors: {
		loadOrders: null,
		loadOrderDetails: null,
		loadReturns: null,
		loadReturnDetails: null,
		loadShippingInfo: null,
	},
};

const ordersReducer = (state = initialState, action = {}) => {
	switch (action.type) {
		case LOAD_ORDERS:
			return {
				...state,
				loadingOrders: true,
			};
		case LOAD_ORDERS_SUCCESS:
			return {
				...state,
				loadingOrders: false,
				orders: action.orders,
				errors: {
					...state.errors,
					loadOrders: null,
				},
			};
		case LOAD_ORDERS_FAIL:
			return {
				...state,
				loadingOrders: false,
				errors: {
					...state.errors,
					loadOrders: action.error,
				},
			};
		case LOAD_ORDER_DETAILS_SUCCESS:
			return {
				...state,
				orderDetails: {
					...state.orderDetails,
					[action.order.orderNumber]: action.order,
				},
				errors: {
					...state.errors,
					loadOrderDetails: null,
				},
			};
		case LOAD_ORDER_DETAILS_FAIL:
			return {
				...state,
				errors: {
					...state.errors,
					loadOrderDetails: action.error,
				},
			};
		case LOAD_RETURNS:
			return {
				...state,
				loadingReturns: true,
			};
		case LOAD_RETURNS_SUCCESS:
			return {
				...state,
				loadingReturns: false,
				returns: action.returns,
				errors: {
					...state.errors,
					loadReturns: null,
				},
			};
		case LOAD_RETURNS_FAIL:
			return {
				...state,
				loadingReturns: false,
				errors: {
					...state.errors,
					loadReturns: action.error,
				},
			};
		case LOAD_RETURN_DETAILS_SUCCESS:
			return {
				...state,
				returnDetails: {
					...state.returnDetails,
					[action.returnOrder.returnId]: action.returnOrder,
				},
				errors: {
					...state.errors,
					loadReturnDetails: null,
				},
			};
		case LOAD_RETURN_DETAILS_FAIL:
			return {
				...state,
				errors: {
					...state.errors,
					loadReturnDetails: action.error,
				},
			};
		case LOAD_SHIPPING_INFO_SUCCESS:
			return {
				...state,
				shippingInfo: {
					...state.shippingInfo,
					[action.shippingInfo.orderNumber]: action.shippingInfo,
				},
				errors: {
					...state.errors,
					loadShippingInfo: null,
				},
			};
		case LOAD_SHIPPING_INFO_FAIL:
			return {
				...state,
				errors: {
					...state.errors,
					loadShippingInfo: action.error,
				},
			};
		case ADD_PROJECT_ORDER:
			const orders = state.orders.map((order) => {
				if (action.order.orderNumber === order.orderNumber) {
					order.selected = true;
				}
				return order;
			});
			return {
				...state,
				orders,
			};
		case SIGN_USER_OUT:
			return initialState;
		default:
			return state;
	}
};

export default ordersReducer;
