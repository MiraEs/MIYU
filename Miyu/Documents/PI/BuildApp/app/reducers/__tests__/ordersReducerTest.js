'use strict';

jest.unmock('../../../app/reducers/ordersReducer');

import ordersReducer from '../ordersReducer';

import {
	LOAD_ORDERS,
	LOAD_ORDERS_SUCCESS,
	LOAD_ORDERS_FAIL,

	LOAD_ORDER_DETAILS_SUCCESS,
	LOAD_ORDER_DETAILS_FAIL,

	ADD_PROJECT_ORDER,

	LOAD_RETURNS,
	LOAD_RETURNS_SUCCESS,
	LOAD_RETURNS_FAIL,

	LOAD_RETURN_DETAILS_SUCCESS,
	LOAD_RETURN_DETAILS_FAIL,

	LOAD_SHIPPING_INFO_SUCCESS,
	LOAD_SHIPPING_INFO_FAIL,
} from '../../constants/constants' ;


import {
	SIGN_USER_OUT,
} from '../../constants/Auth' ;

describe('ordersReducer reducer', () => {

	it('should return initialState', () => {
		expect(ordersReducer(undefined, {})).toMatchSnapshot();
	});


	it('should LOAD_ORDERS', () => {
		const action = {
			type: LOAD_ORDERS,
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOAD_ORDERS_SUCCESS', () => {
		const action = {
			type: LOAD_ORDERS_SUCCESS,
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOAD_ORDERS_FAIL', () => {
		const action = {
			type: LOAD_ORDERS_FAIL,
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOAD_ORDER_DETAILS_SUCCESS', () => {
		const action = {
			type: LOAD_ORDER_DETAILS_SUCCESS,
			order: {
				ordernumber: '1234',
			},
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOAD_ORDER_DETAILS_FAIL', () => {
		const action = {
			type: LOAD_ORDER_DETAILS_FAIL,
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should ADD_PROJECT_ORDER', () => {
		const action = {
			type: ADD_PROJECT_ORDER,
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOAD_RETURNS', () => {
		const action = {
			type: LOAD_RETURNS,
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOAD_RETURNS_SUCCESS', () => {
		const action = {
			type: LOAD_RETURNS_SUCCESS,
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOAD_RETURNS_FAIL', () => {
		const action = {
			type: LOAD_RETURNS_FAIL,
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOAD_RETURN_DETAILS_SUCCESS', () => {
		const action = {
			type: LOAD_RETURN_DETAILS_SUCCESS,
			returnOrder: {
				returnId: 2,
			},
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOAD_RETURN_DETAILS_FAIL', () => {
		const action = {
			type: LOAD_RETURN_DETAILS_FAIL,
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOAD_SHIPPING_INFO_SUCCESS', () => {
		const action = {
			type: LOAD_SHIPPING_INFO_SUCCESS,
			shippingInfo: {
				orderNumber: '1234',
			},
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOAD_SHIPPING_INFO_FAIL', () => {
		const action = {
			type: LOAD_SHIPPING_INFO_FAIL,
		};
		const state = ordersReducer(undefined, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should SIGN_USER_OUT', () => {
		const action = {
			type: SIGN_USER_OUT,
		};
		const state = ordersReducer(undefined, action);
		expect(
			state,
		).toMatchSnapshot();
	});

});
