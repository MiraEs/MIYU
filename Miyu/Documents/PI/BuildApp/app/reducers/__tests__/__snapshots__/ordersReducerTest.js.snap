// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ordersReducer reducer should ADD_PROJECT_ORDER 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": true,
  "loadingReturns": true,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should LOAD_ORDER_DETAILS_FAIL 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": undefined,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": true,
  "loadingReturns": true,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should LOAD_ORDER_DETAILS_SUCCESS 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": true,
  "loadingReturns": true,
  "orderDetails": Object {
    "undefined": Object {
      "ordernumber": "1234",
    },
  },
  "orders": Array [],
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should LOAD_ORDERS 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": true,
  "loadingReturns": true,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should LOAD_ORDERS_FAIL 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": undefined,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": false,
  "loadingReturns": true,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should LOAD_ORDERS_SUCCESS 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": false,
  "loadingReturns": true,
  "orderDetails": Object {},
  "orders": undefined,
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should LOAD_RETURN_DETAILS_FAIL 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": undefined,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": true,
  "loadingReturns": true,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should LOAD_RETURN_DETAILS_SUCCESS 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": true,
  "loadingReturns": true,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {
    "2": Object {
      "returnId": 2,
    },
  },
  "returns": Array [],
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should LOAD_RETURNS 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": true,
  "loadingReturns": true,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should LOAD_RETURNS_FAIL 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": undefined,
    "loadShippingInfo": null,
  },
  "loadingOrders": true,
  "loadingReturns": false,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should LOAD_RETURNS_SUCCESS 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": true,
  "loadingReturns": false,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {},
  "returns": undefined,
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should LOAD_SHIPPING_INFO_FAIL 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": undefined,
  },
  "loadingOrders": true,
  "loadingReturns": true,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should LOAD_SHIPPING_INFO_SUCCESS 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": true,
  "loadingReturns": true,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {
    "1234": Object {
      "orderNumber": "1234",
    },
  },
}
`;

exports[`ordersReducer reducer should SIGN_USER_OUT 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": true,
  "loadingReturns": true,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {},
}
`;

exports[`ordersReducer reducer should return initialState 1`] = `
Object {
  "errors": Object {
    "loadOrderDetails": null,
    "loadOrders": null,
    "loadReturnDetails": null,
    "loadReturns": null,
    "loadShippingInfo": null,
  },
  "loadingOrders": true,
  "loadingReturns": true,
  "orderDetails": Object {},
  "orders": Array [],
  "returnDetails": Object {},
  "returns": Array [],
  "shippingInfo": Object {},
}
`;
